New tailwind-based joinlemmy-site

dessalines opened the pull request:
You can test this at http://jointest.lemmy.ml:1234/

TODOs yet:

- [x] Update pictures
- [x] Alter specific texts to allow emphasis (make sure you're using interpolation correctly, and if possible get rid of the `<T` component)
- [x] Add donation links for apps and instances.
- [x] Add a funding drive type box to the support page.
- [x] Categories, and popup instance onboarding
- [x] Add 3 other major UIs linked [here](https://github.com/LemmyNet/lemmy-ansible/issues/180)
- [x] Update donors

SorteKanin commented:
Hey, super glad to hear the joinlemmy site is being worked on. I think there's a lot of room for improvements. I'm not really a frontend guy on the technical side, but I'd love to help with changing some of the phrasing or "style" of presenting Lemmy (currently I think it is way too technical, for one thing).

What's the "guiding principles"/primary motivation in this rework? Is this primarily a change of the visual design or is changing the content also included? I personally think it would make sense to do a complete makeover while we're at it anyway.

Have you considered including a server covenant similar to (not necessarily equal to) [Mastodon's](https://joinmastodon.org/covenant) and requiring it for an instance to appear on the instance list?

Nutomic commented:
@SorteKanin This is only changing the design while keeping current texts, we want to change them later. If you want to help with that, feel free to make commits to the [joinlemmy-translations repo](https://github.com/LemmyNet/joinlemmy-translations/blob/1bc69869fda7ee144ddfbc4d9fb29af3a0d4619e/translations/en.json).

A server covenant would be an entirely different matter. Personally Id rather not do that, because it means a lot of extra work to verify that instances follow it.

dessalines commented:
> Is this primarily a change of the visual design or is changing the content also included? I personally think it would make sense to do a complete makeover while we're at it anyway.

Yes this is mainly a visual layout design, not altering the language. The other things you could open up as separate issues.

SorteKanin commented:
>This is only changing the design while keeping current texts, we want to change them later. If you want to help with that, feel free to make commits to the [joinlemmy-translations repo](https://github.com/LemmyNet/joinlemmy-translations/blob/1bc69869fda7ee144ddfbc4d9fb29af3a0d4619e/translations/en.json).

Alright, although often the text and design follow each other to some degree. Ideally it should be designed as a whole. I don't mean to compare everything to mastodon (I don't even use mastodon myself), but [their equivalent site](https://joinmastodon.org/) looks a lot more inviting (and not just because of the artwork).

>Personally Id rather not do that, because it means a lot of extra work to verify that instances follow it.

Have you thought about spending some more time on recruiting contributors? Especially people who are not necessarily coders? The work of verifying that servers follow a covenant requires no coding skills, for instance. I can recommend [this talk on how to look for non-coding people in open source](https://www.youtube.com/watch?v=u3PJaiSpbmc). A lot of it is just asking and empowering people to do it!

Spending more time on recruiting contributors could be a much better use of time on the long term.

dessalines commented:
These are larger issues that have nothing to do with this PR. Please open these up as separate issues elsewhere, or they will get lost.

Nutomic commented:
Looks good, some feedback:
- Try blurring the screenshots on index page so people dont try to read that text
- Clicking on the dot buttons to switch screenshots shouldnt change scroll position
- Add some screenshots with different themes and frontends
- Nice job with the "Join a server" dialog! 
    - The buttons are using radio button design, would make more sense to use normal buttons. Or simply get rid of circles on the right and add an outline for each clickable area
    - Step 3 "Join" is unused, could remove it
    - If you select an interest, then close the dialog and open it again, it goes directly to languages. Should always start from interests after you click the join button
- I would also add a button "Run a server" at the top of the page for people who are searching that option (under "A link aggregator for the fediverse")
- In the "More features" section, icons appear clickable based on mouse cursor, but nothing happens on click
- In the same section, green text from links and code is too visible. I would change the code color to grey/black, and instead of inline links consider making the whole box clickable
- The language and activity sorts for instance page look very useful
- How are the categories for instance list defined? Some of them are completely empty, while "General" has dozens of items
- Instead of "Join a server" I would rather put "Browse" or "Explore". Basically let people know that they are free to look around and arent required to create an account
- Add tooltips for the users / posts / monthly users badges to indicate what they mean
- State explicitly that users can interact with content on other servers so its not very important which instance they pick (in top text)
- On News page, every item has simply "News" as title. Should read the actual title from first line of markdown
- All items have @nutomic and @dessalines which is redundant, might as well leave the names out
- Text preview contains unrendered markdown, should only take the actual text
- On apps page the headings like "Mobile apps for Android." should be bigger and/or different color
- On donation page, all the names of donors, contributors, translators should be updated automatically because we will never get around to do it manually. Otherwise get rid of them
- Move crypto donation options into the box at the top of the page. To save space it can be done with a popup or expanding section
- The names of contributors and donors link to different websites including github, lemmy and mastodon. Better make it consistent and always link to the same site, or get rid of links
- The donation amount of €16,667 is the monthly goal? Needs to be stated explicitly. Also not clear what "14 months remaining" means
- "Your donations" link points to the same page
- I would use "Donate" instead of "Support" everywhere in the top box so its clear what we are talking about. Also simplify the heading to "Donate"
- On bottom right instead of "All rights reserved", mention that the site is open source under AGPL and link to git repo. 

That should be enough for now :laughing:

SorteKanin commented:
Generally an improvement I would say. Some feedback:

### Dark theme
I think the dark theme is the wrong default choice. Dark themed websites are attractive to programmers and developers, but most people find white themes to be the default and more familiar/welcoming option. Almost all large websites use a white theme by default (some even don't have dark theme options). Even Mastodon uses a white theme for joinmastodon.org (even if the top graphic is quite dark).

Considering Lemmy is already _heavily_ dominated by developers and other tech-savvy people, I think it would be better to use a light theme that would be more familiar/welcoming to non-tech people.

### Run a server button

>I would also add a button "Run a server" at the top of the page for people who are searching that option

I disagree. Those who are looking for that option won't have trouble finding it and 99.9% (if not more) of users are not going to be looking for that option.

### Lemmy screenshots on home page

I think the screenshots are too big - at first I actually thought it was part of the site's content because the text was the same size or larger than the actual content. It's also not obvious that there is anything below the images that the user is meant to scroll to. If the images were smaller, it'd be clearer that it's just an example screenshot and there is more stuff below the screenshots.

### Features highlights
Here I'm talking about the "Open Source", "Blazing Fast", "Powerful Mod Tools", "Censorship resistant" and "Federation" boxes with associated images.

First of all (and this goes into kind of what I said above, so maybe more about content than design), I don't think most users will care about the first two. Many users will not even know what "open source" means. Maybe instead highlighting that Lemmy is *non-profit*, i.e. not based on a for-profit corporation like traditional social media would be a better way to explain this aspect.

As for the blazing fast one: I think most users don't consciously care about speed - they will _expect speed by default_ but they are not "impressed" by speed. Speed is only a concern if it is missing, but it is not a feature worth highlighting.

I'm not sure it's accurate to highlight "Censorship resistance" as a feature, as the vast majority of users won't run their own instance (most will not even know how to) and therefore won't be in control of their own content in this way (and probably most users do not desire to be in total control of their content as they would rather leave moderation to others; this is not a bad thing).

One last point: The images in this section are quite bland and "gray", almost boring. They don't really pop out to the eyes. I would prefer some more colourful images.

### Server browser

I love the new server browser and sorting options, this is a massive improvement. Maybe the physical location of the server/legal entity of the instance could also be taken into account, like Mastodon does? Not sure how to get that data though.

dessalines commented:
Okay I've updated it based on comments. Check out http://jointest.lemmy.ml:1234 again.

> Try blurring the screenshots on index page so people dont try to read that text

I tried it but even with a tiny blur, it looks almost dizzying. I spose that's why a lot of sites use `Lorem ipsum...` replacers. I'll see if there's some plugin I can find for that.

> Clicking on the dot buttons to switch screenshots shouldnt change scroll position

That one I can't mess with easily, because its non-javascript from the daisyUI carousel. I *could* try to build one myself, but it seems really minor.

> Add some screenshots with different themes and frontends

I spose I could do this, as long as they're further along in the carousel. I hate adding images to the git repo tho, it bloats it massively when we want to change / remove them. I tried adding a few of the phone ones, but unfortunately the carousel needs them to be the same aspect ratio.

> The buttons are using radio button design, would make more sense to use normal buttons. Or simply get rid of circles on the right and add an outline for each clickable area

I refactored this to smart wrapping flex row of buttons

> Step 3 "Join" is unused, could remove it

Done.

> If you select an interest, then close the dialog and open it again, it goes directly to languages. Should always start from interests after you click the join button

Fixed.

> I would also add a button "Run a server" at the top of the page for people who are searching that option (under "A link aggregator for the fediverse")

I looked at joinmastodon (they only show a join mastodon.org, and a show all instances box), and decided to get rid of that from the very top (it is still below in the *create your own platform* block). I agree with @SorteKanin there, that we should be user-oriented at the very top at least. Those who want to run their own can check the docs, or further down.

> In the "More features" section, icons appear clickable based on mouse cursor, but nothing happens on click

Fixed.

> In the same section, green text from links and code is too visible. I would change the code color to grey/black, and instead of inline links consider making the whole box clickable

I changed all these to regular links so they look more muted now, but I'd rather not mess with the i18n strings for those, especially since one text blob might have several links.


> How are the categories for instance list defined? Some of them are completely empty, while "General" has dozens of items

[I had to manually define them as recommended instances unfortunately](https://github.com/LemmyNet/joinlemmy-site/pull/243/files#diff-f78556500c68f47e12256acdcb2e85d0aa6847ba207adfd7344e2be7cce56af1R119), but at least everything is well typed. So far I just have domains have an array of languages and topics. But we could add more filters later. The `all` filters will ignore that explicit list of recommended instances.

There's also the legit concern that perhaps we should **only show** results that we've explicitly included (in that recommended_instances array), since the crawler might scan pedo stuff like happened before. It might be okay to do that, because people will see that their instance isn't shown on join-lemmy.org, and then realize they need to make a PR to categorize their instance.

> Add tooltips for the users / posts / monthly users badges to indicate what they mean

Done.

> State explicitly that users can interact with content on other servers so its not very important which instance they pick (in top text)

Fixed, and I also changed up the stats a bit.

> On News page, every item has simply "News" as title. Should read the actual title from first line of markdown

Fixed, was a code glitch.

> All items have @Nutomic and @dessalines which is redundant, might as well leave the names out

They were from the mockup, but I've removed them.

> Text preview contains unrendered markdown, should only take the actual text

Fixed, I render the markdown rather than try to alter it.

> On apps page the headings like "Mobile apps for Android." should be bigger and/or different color

I made the titles bigger here.

> On donation page, all the names of donors, contributors, translators should be updated automatically because we will never get around to do it manually. Otherwise get rid of them

This one I *really* don't want to handle right now.  All 3 donation platforms have their own unique API (and some don't even have one), and it'd take a ton of work to get API keys, learn their APIs, create a separate program that writes all these out and integrates them... when it really only takes me ~ 15 minutes (ideally once a month, when I have to update the total counts anyway), to look at the top donors for each. 

The translators comes from `lemmy-translations/translators.json` , which requires a manual PR there, but I don't want to uncredit / remove the section thanking them. 

I'd def be willing to get rid of the general sponsors section, because we have > 1k supporters currently, and that'd be way too much to put on the page anyway. I eventually just started to give up when I did this manually. But the sponsors above silver is < 20 ppl.

> Move crypto donation options into the box at the top of the page. To save space it can be done with a popup or expanding section

I moved it to its own page now.

> The names of contributors and donors link to different websites including github, lemmy and mastodon. Better make it consistent and always link to the same site, or get rid of links

I changed a few of them to github ones. The translators links were made by them also. A lot of other ones are patreon links. I don't think its necessary to have them all be from the same platform anyway, its up to them.

> The donation amount of €16,667 is the monthly goal? Needs to be stated explicitly. Also not clear what "14 months remaining" means

I removed that number, and replaced it with the **devs** funded, since that's more important anyway. 

The months remaining was because this could be a yearly funding drive, and its good to have a deadline there (I set it to one year from this december currently, but we can change that to whenever). Let me know if maybe a tooltip would help there.

> "Your donations" link points to the same page

That same component is used on both the main and donate pages, so it will navigate to the donate page if you click it from main. Either way its harmless.

> I would use "Donate" instead of "Support" everywhere in the top box so its clear what we are talking about. Also simplify the heading to "Donate"

Fixed. I did keep the `Support on Liberapay` buttons cause that looks a little better imo. Otherwise I can just get rid of "Support on" and keep the site name.

> On bottom right instead of "All rights reserved", mention that the site is open source under AGPL and link to git repo.

Done

---

> Dark theme

I'm not totally opposed... I randomly tried some different daisyUI themes for fun, and they didn't look terrible, but I'd need to change some of the background classes I've been using. We'll have to revisit at another time bc I already did so much work on this dark-specific theme.

> Lemmy screenshots on home page

I made them take less width, and also moved the follow block above the carousel. It looks much better, thx!

> Features highlights

I agree these are a bit "dev-oriented", and the images are a bit bland. This is another "sunk cost" thing for me rn tbh, it'd take a lot of work to create new images and rework all that text.

I did make the cards use that pretty gradient tho now.

> Maybe the physical location of the server/legal entity of the instance could also be taken into account, like Mastodon does?

We *could* add that to the instance definitions, it'd just be a lot of manual work. Also I'm not sure how to break up *regions*, which can get infinitely specific, into selectable filters. I'm not against it completely, I just don't know the best way to do that.

wiki-me commented:
Did the "recommended" section of the instances was removed?  This can create a problem of [choice overload](https://en.wikipedia.org/wiki/Overchoice).

Also it mentioned:

>You can access all content in the lemmyverse from any server, so it doesn't matter which one you choose.

That's not true, e.g. in lemmy world you can't accesss lemmygrad because it is blocked. this might cause confusion and will make people give up on the platform.

I would suggest "don't worry too much about which server to pick, There are tools for migrating relatively easily to different one".

I also don't feel like the elevator pitch for donating is good enough, why should  users care about full time development?, I suggest something like "more full time developers will help us make lemmy the best platform it can be, and attract more awesome content creators". A link to some road map could also be beneficial (donating should become a "[conditioned reward](https://en.wikipedia.org/wiki/Reward_system)".

wikipedia does [A/B testing](https://en.wikipedia.org/wiki/A/B_testing) for the fundraising pitches, maybe lemmy should do them as well, even asking the communities for ideas because this is something that could have a big impact.

SleeplessOne1917 commented on src/shared/components/navbar.tsx:
```diff
@@ -1,69 +1,126 @@
+  <>
+    <NavLink
+      content={
+        <Link onClick={() => closeNavbarDropdown()} to="/instances">
```
Since `closeNavbarDropdown` takes no args, you can just pass it in directly instead of calling it in a lambda function, i.e.:
```tsx
<Link onClick={closeNavbarDropdown} to="/instances">
```

You can do that with the other nav links as well.

SleeplessOne1917 commented on src/shared/components/navbar.tsx:
```diff
@@ -1,69 +1,126 @@
+    />
+    <NavLink
+      content={
+        <a onClick={() => closeNavbarDropdown()} href={`/docs/index.html`}>
```
If you're using a string literal for the `href`, you don't need the curly braces.

SleeplessOne1917 commented on src/shared/components/main.tsx:
```diff
@@ -1,12 +1,432 @@
+
+const FeatureCard = ({ pic, title, subtitle, classes }) => (
+  <div className={`card ${CARD_GRADIENT} shadow-xl ${classes}`}>
+    <figure className="p-4">
+      <img src={pic} className="rounded-xl w-full object-fill min-h-[300px]" />
+    </figure>
```
Why use figure if you're not including a caption? I think the `img` tag alone should work fine in that case.

SleeplessOne1917 commented:
Some feedback on the UI:

- It would be nice to put a [scroll-smooth](https://tailwindcss.com/docs/scroll-behavior) class on the carousel to make the automatic scrolling a bit less jarring. Also, the the carousel doesn't move when you go from the 1st or 3rd section to the 2nd one.
- Having the top and bottom nav bars almost exactly mirror each other looks kind of strange.
- There are a couple more notable frontends that could be added to the apps page, such as Liftoff, Thunder, Eternity, Boost, and Sync.

dessalines commented:
> It would be nice to put a [scroll-smooth](https://tailwindcss.com/docs/scroll-behavior) class on the carousel to make the automatic scrolling a bit less jarring. Also, the the carousel doesn't move when you go from the 1st or 3rd section to the 2nd one on larger screen sizes.

I fixed the smooth scroll thing, but can't mess with anything else too much. daisyUI's carousel is pretty limited.

> Having the top and bottom nav bars almost exactly mirror each other looks kind of strange.

That's fair, but at least you shouldn't normally see them at the same time.

> There are a couple more notable frontends that could be added to the apps page, such as Liftoff, Thunder, Eternity, Boost, and Sync.

I'll add some of these shortly. I'm still somewhat torn on including closed source ones.

dessalines commented:
Updated on the linked site again.

wiki-me commented:
> I'm still somewhat torn on including closed source ones.

Don't think we need to do that, what if one of those closed source apps does something really shaddy, if you are willing to use something closed source we can link to the "lemmy client" tag [on alternativeto](https://alternativeto.net/tag/lemmy-client/).

dessalines commented:
I'll add a warning disclaimer to those ones.

dessalines commented:
> I also don't feel like the elevator pitch for donating is good enough, why should users care about full time development?, I suggest something like "more full time developers will help us make lemmy the best platform it can be, and attract more awesome content creators". A link to some road map could also be beneficial (donating should become a "[conditioned reward](https://en.wikipedia.org/wiki/Reward_system)".

Sry I didn't address these ones. I agree that line could be tweaked. How about instead of 

> "Your donations directly support full-time development of the project."

We have

> "Your donations will help grow the Lemmy developer co-op, and ensure that we can keep adding new features, and improving Lemmy.

I'm up for other suggestions there tho.

Die4Ever commented:
The preview site looks really great! IDK if I should make a separate issue or not, but I would like the ability to hotlink to the "What topic are you most interested in?" questionnaire for choosing an instance to join.

wiki-me commented:
Not sure the average joe even knows what a co-op (or a cooperative ) is, so the impression will be at best neutral and not positive , switching co-op for "non profit" is probably better.

Adding new features and improving lemmy feels a little too abstract, maybe adding "user requested" is better:

    "Your donations will help grow the Lemmy developer non profit, and ensure that we can keep adding new features users are requesting, and improving Lemmy."

I feel my original purposal was better, but what you and i feel is not really indictive of what the average user will feel, if A/B testing is not an option (but i really think it is worth the effort) i think even just making two posts on lemmy (one asking for options and another asking for people to vote on them) is a better option then what people on github think because it is not a representative sample, the point is not to find what people here think is good, but what will maximize the [expected value](https://en.wikipedia.org/wiki/Expected_value) the donations.

[misskey](https://misskey-hub.net/en/) , [inkscape](https://inkscape.org/support-us/donate/) , and [blender](https://fund.blender.org/?utm_medium=www-nav) (which seem to do well in term of funding) use the words "awesome" and "support" which feel more meaningful at least to me (but we might turn those words into cliches).

Nutomic commented:
> Instead of "Join a server" I would rather put "Browse" or "Explore". Basically let people know that they are free to look around and arent required to create an account

This is fixed in the instance list, though the more information window still has "join" buttons.

> [I had to manually define them as recommended instances unfortunately](https://github.com/LemmyNet/joinlemmy-site/pull/243/files#diff-f78556500c68f47e12256acdcb2e85d0aa6847ba207adfd7344e2be7cce56af1R119), but at least everything is well typed. So far I just have domains have an array of languages and topics. But we could add more filters later. The all filters will ignore that explicit list of recommended instances.
> 
> There's also the legit concern that perhaps we should only show results that we've explicitly included (in that recommended_instances array), since the crawler might scan pedo stuff like happened before. It might be okay to do that, because people will see that their instance isn't shown on join-lemmy.org, and then realize they need to make a PR to categorize their instance.

Hmm this is tricky. On one hand its not much to ask to make a small file edit to get the instance included. On the other hand there will be many cases where this wont work out in practice, eg unable to access Github for different reasons, not enough tech literacy to figure out the edit steps, language barrier when reading instructions. Also most people will tag their instances as "General", and it wont be easy to figure out when new tags should be added. We could handle this by adding a category setting for Lemmy which can be read over the API. Languages could also be read from `/api/v3/site` `discussion_languages`, but in practice there are many instances with all languages active. So I think to make it work with info defined in joinlemmy repo, we will need some community members to manage instance attributes.

> > Text preview contains unrendered markdown, should only take the actual text
> 
> Fixed, I render the markdown rather than try to alter it.

The text preview on `/news` is still pretty useless as it repeats the title, and then "written by". Might be better to strip out the first few lines of markdown so it starts from the actual text body. But then each one will preview the generic "what is lemmy" block. Maybe get rid of the text preview but then it will look too empty...



> > On donation page, all the names of donors, contributors, translators should be updated automatically because we will never get around to do it manually. Otherwise get rid of them
>
> This one I really don't want to handle right now. All 3 donation platforms have their own unique API (and some don't even have one), and it'd take a ton of work to get API keys, learn their APIs, create a separate program that writes all these out and integrates them... when it really only takes me ~ 15 minutes (ideally once a month, when I have to update the total counts anyway), to look at the top donors for each.
> 
The translators comes from lemmy-translations/translators.json , which requires a manual PR there, but I don't want to uncredit / remove the section thanking them.
>
> I'd def be willing to get rid of the general sponsors section, because we have > 1k supporters currently, and that'd be way too much to put on the page anyway. I eventually just started to give up when I did this manually. But the sponsors above silver is < 20 ppl.

Hmm so everything on `/donate` page is manually hardcoded? I was expecting at least the total monthly amount was automatically requested from each platform API so it stays up to date during funding drive. The name list is less important, but I hope you will remember to update it regularly because previously it wasnt updated at all for some years. And yes if general sponsors section is missing hundreds of people, get rid of it for now and we can consider readding once its automated. 

> >  Features highlights
> 
> I agree these are a bit "dev-oriented", and the images are a bit bland. This is another "sunk cost" thing for me rn tbh, it'd take a lot of work to create new images and rework all that text.

I would leave the texts for now. When launching the new design we can ask the community to help improve the texts and submit better images.

> > Maybe the physical location of the server/legal entity of the instance could also be taken into account, like Mastodon does?
>
> We could add that to the instance definitions, it'd just be a lot of manual work. Also I'm not sure how to break up regions, which can get infinitely specific, into selectable filters. I'm not against it completely, I just don't know the best way to do that.

You could consider using geoip to find the country where a server is hosted, and then showing the country flag. A bit like [the-federation.info](https://the-federation.info/platform/73). Should be pretty easy to implement but detection will be wrong in some cases.

---

> Did the "recommended" section of the instances was removed? This can create a problem of [choice overload](https://en.wikipedia.org/wiki/Overchoice).

This might not be necessary anymore as there are filters for topic and language which can serve the same purpose. However the language filter is incomplete, feddit.de is missing for German. Portuguese and Portuguese (Brazil) should be merged. Other languages like Chinese or Esperanto have no instances so they should be removed from the dropdown.

> >  You can access all content in the lemmyverse from any server, so it doesn't matter which one you choose.
>
> That's not true, e.g. in lemmy world you can't accesss lemmygrad because it is blocked. this might cause confusion and will make people give up on the platform.
>
> I would suggest "don't worry too much about which server to pick, There are tools for migrating relatively easily to different one".

Its tricky to describe this concisely and accurately. I propose: 

Lemmy has many servers which are operated independently, with their own rules, topics and languages. Use the list below to find one that best matches your interests. Thanks to federation you can freely interact with content from other instances, so don't worry too much which one to pick. You can also migrate between servers, or register multiple accounts on different instances to find the one you like most. Read the [documentation](https://join-lemmy.org/docs/index.html) for more details.

And for donations:

Lemmy is free, open-source software, meaning no advertising, monetizing, or venture capital, ever. However full-time developers are necessary to maintain the project, fix bugs and implement important features. The developers are paid exclusively paid from donations. So please consider donating if you want Lemmy to keep improving. More donations means that more developers can be paid, and improvements can be implemented faster.

Also agree about making a post to get suggestions for these two, and for general improvements to joinlemmy texts. We can do that together with the announcement of the redesign, or right afterwards.

testman42 commented:
Maybe choose a bit darker shade of green for the buttons, so that the white text on them can be read more easily.

wiki-me commented:
> This might not be necessary anymore as there are filters for topic and language which can serve the same purpose. However the language filter is incomplete, feddit.de is missing for German. Portuguese and Portuguese (Brazil) should be merged. Other languages like Chinese or Esperanto have no instances so they should be removed from the dropdown.

I think 95% percent of users will want a generic instance in english, i would be careful about this, when the reddit fiasco happened i constantly read complains about picking an instance.

> Lemmy has many servers which are operated independently, with their own rules, topics and languages. Use the list below to find one that best matches your interests. Thanks to federation you can freely interact with content from other instances, so don't worry too much which one to pick. You can also migrate between servers, or register multiple accounts on different instances to find the one you like most. Read the [documentation](https://join-lemmy.org/docs/index.html) for more details.

I like my suggestion because it is shorter  , according to [this](https://www.nngroup.com/articles/how-little-do-users-read/) (see graph) they won't read somwhere about half of what is on that page.

> Lemmy is free, open-source software, meaning no advertising, monetizing, or venture capital, ever. However full-time developers are necessary to maintain the project, fix bugs and implement important features. The developers are paid exclusively paid from donations. So please consider donating if you want Lemmy to keep improving. More donations means that more developers can be paid, and improvements can be implemented faster.

the part about "full-time developers are necessary to maintain the project" might get a push back, people complained about wikipedia and the whole "we need donations to survive" when they got a lot of money and that is why now they do  community reviews for these types of texts.

> Also agree about making a post to get suggestions for these two, and for general improvements to joinlemmy texts. We can do that together with the announcement of the redesign, or right afterwards.

Yeah otherwise we might argue over feelings, all the options suggested here seem fine for a short period (1-2 weeks), and this PR is already pretty big.

dessalines commented:
> but I would like the ability to hotlink to the "What topic are you most interested in?" questionnaire for choosing an instance to join.

fixed. That button now is a pseudo `<a` element that will let you click and hold to copy its link.

> This is fixed in the instance list, though the more information window still has "join" buttons.

Fixed.

> So I think to make it work with info defined in joinlemmy repo, we will need some community members to manage instance attributes.

I think that's the best way to handle this; ask instances to categorize themselves. Otherwise it will be our own curated list.

> The text preview on /news is still pretty useless as it repeats the title, and then "written by". Might be better to strip out the first few lines of markdown so it starts from the actual text body. But then each one will preview the generic "what is lemmy" block. Maybe get rid of the text preview but then it will look too empty...

This uses either the title from the markdown file, or if that's missing, the first few lines of markdown. But I've made it start from the 3rd line, which looks a little better.

> Hmm so everything on /donate page is manually hardcoded? I was expecting at least the total monthly amount was automatically requested from each platform API so it stays up to date during funding drive. The name list is less important, but I hope you will remember to update it regularly because previously it wasnt updated at all for some years. 

Yeah unfortunately, but I'll set a reminder to update this monthly.

> And yes if general sponsors section is missing hundreds of people, get rid of it for now and we can consider readding once its automated.

Done. But I don't see myself automating this in the future, because of the large work involved in having to deal with 3 separate APIs.

Feddit.de is defined, but the last crawler run didn't pick it up for some reason. I'll add it to recommended just in case.

Die4Ever commented:
yep this is cool http://jointest.lemmy.ml:1234/?showJoinModal=true

definitely gonna be my preferred way to get people on Lemmy, thanks!

aeharding commented:
First off, very nice work. This is not just an improvement - it's a great design.

I was wondering, would it makes sense to put a "sign up" button next to the "browse" of each instance? Or perhaps even replace "browse" with "sign up"? (Given most users will probably be using this flow to sign up.)

I know it's just one more click - but looking at Join Mastodon, "create account" is the *only* action when browsing instances.

(Also as a sidenote: I link to join-lemmy.org in-app from Voyager for users to sign up, and it would be great if the flow was `pick instance -> signup` vs `pick instance -> browse -> signup`.)

wiki-me commented:
Am i the only one who feel like the site would look better if the images would be removed? feel free to give a "thumbs up" or "thumbs down" so we could get some indication if that is the common sentiment .

dessalines commented:
> I was wondering, would it makes sense to emphasize signing up vs browsing? (Given most users will probably be using this flow to sign up.)

Good idea, I've got this added now.

As far as the images, I agree they could be better, but they can be changed / added at a later time. I'd accept submissions as long as they maintain the same aspect ratios.
